http-client/src/main/java/io/airlift/http/client/jetty/JettyHttpClient.java

```diff
@@ -165,7 +169,12 @@ public JettyHttpClient(
         creationLocation.fillInStackTrace();
 
         SslContextFactory sslContextFactory = new SslContextFactory();
-        sslContextFactory.setEndpointIdentificationAlgorithm("HTTPS");
+        if (config.getHostnameVerificationEnabled()) {
+            sslContextFactory.setEndpointIdentificationAlgorithm("HTTPS");
+        }
+        else {
+            sslContextFactory.setEndpointIdentificationAlgorithm(null);
```

Comment:
The `HTTPS` just enables the validation that the hostname matches the ssl cert. Looking around, it looks like the only valid options are "HTTPS", "LDAPS", or null. (https://docs.oracle.com/javase/8/docs/technotes/guides/security/StandardNames.html#h3501)

TIL